en/docs/deploy-and-publish/deploy-on-gateway/choreo-connect/analytics/pulbish-custom-analytics-attributes.md

```diff
@@ -0,0 +1,134 @@
+# Publish Custom Attributes With Analytics
+
+Choreo Connect provides default set of attributes with the analytics data when an API invocation happens. Additional to that, users
+can add custom analytics data with an analytics event due to this feature. If there are specific request headers, response
+headers, response trailers relevant to the event, users can retrieve those values and use with Chore Connect analytics.
+This guide will explain the steps required to do it.
+
+!!! note
+    **Request headers** (headers sent to the upstream), **Response headers** (headers coming from the upstream) and **Response trailers** can only use with this feature.
+
+## Creating The Custom Analytics Data Provider JAR
+
+You need to create a new `Java Maven` project to obtain this JAR.
+
+There is an existing sample project in [here](https://github.com/wso2/product-microgateway/tree/main/samples/analytics-custom-data-provider).
+If you wish to use that sample instead of developing the sample from scratch, then you can ignore the steps of creating
+the sample and start from [here]({{base_path}}/deploy-on-gateway/choreo-connect/analytics/pulbish-custom-analytics-attributes/#build-the-project-jar-addition-to-the-enforcer).
+
+### Configuring the pom.xml file
+
+1. Add the wso2-nexus repository to the pom.xml file.
+
+    ```xml
+    <repository>
+        <id>wso2-nexus</id>
+        <name>WSO2 internal Repository</name>
+        <url>https://maven.wso2.org/nexus/content/groups/wso2-public/</url>
+        <releases>
+            <enabled>true</enabled>
+            <updatePolicy>daily</updatePolicy>
+            <checksumPolicy>ignore</checksumPolicy>
+        </releases>
+    </repository>
+    ```
+2. Add relevant dependencies.
+
+    ```xml
+    <dependency>
+        <groupId>org.wso2.carbon.apimgt</groupId>
+        <artifactId>org.wso2.carbon.apimgt.gateway</artifactId>
+        <version>${carbon.apimgt.version}</version>
+    </dependency>
+    <dependency>
+        <groupId>org.wso2.carbon.apimgt</groupId>
+        <artifactId>org.wso2.carbon.apimgt.common.analytics</artifactId>
+        <version>${carbon.apimgt.version}</version>
+    </dependency>
+    ```
+
+!!! Info
+
+	- If you need above mentioned JARs, you can find them from [here](https://github.com/wso2/product-microgateway/tree/main/samples/analytics-custom-data-provider/jars).
+    - When working with above sample project, you have to follow the below steps to add the required artifacts to the local m2 repository and point it as a repository in pom file.
+        1. Find the `org.wso2.carbon.apimgt.gateway` and `org.wso2.carbon.apimgt.common.analytics` jars from [here](https://github.com/wso2/product-microgateway/tree/main/samples/analytics-custom-data-provider/jars).
```

Comment:
No such file in the repo. From the gitignore we do not add jars anyway. 
Is that the same approach used in api manager docs? IMO, it is not common to have jars uploaded in git and ask the users to use them.